gtfs/spec/en/reference.md

```diff
@@ -94,10 +94,10 @@ This specification defines the following files:
 |  [stop_times.txt](#stop_timestxt)  | **Required** | Times that a vehicle arrives at and departs from stops for each trip. |
 |  [calendar.txt](#calendartxt)  | **Conditionally Required** | Service dates specified using a weekly schedule with start and end dates. <br><br>Conditionally Required:<br> - **Required** unless all dates of service are defined in [calendar_dates.txt](#calendar_datestxt).<br> - Optional otherwise. |
 |  [calendar_dates.txt](#calendar_datestxt)  | **Conditionally Required** | Exceptions for the services defined in the [calendar.txt](#calendartxt). <br><br>Conditionally Required:<br> - **Required** if [calendar.txt](#calendartxt) is omitted. In which case [calendar_dates.txt](#calendar_datestxt) must contain all dates of service. <br> - Optional otherwise. |
-|  [fare_attributes.txt](#fare_attributestxt)  | **Conditionally Forbidden** | Fare information for a transit agency's routes.<br><br>Conditionally Forbidden:<br>- **Forbidden** if [fare_leg_rules.txt](fare_leg_rulestxt) is defined.<br>- Optional otherwise. |
-|  [fare_rules.txt](#fare_rulestxt)  | **Conditionally Forbidden** | Rules to apply fares for itineraries.<br><br>Conditionally Forbidden:<br>- **Forbidden** if [fare_leg_rules.txt](fare_leg_rulestxt) is defined.<br>- Optional otherwise. |
-|  [fare_leg_rules.txt](#fare_leg_rulestxt)  | **Conditionally Forbidden** | Fare rules for individual legs of travel.<br><br>File fare_leg_rules.txt provides a more detailed method for modeling fare structures. As such, the use of fare_leg_rules.txt is entirely seperate from files fare_attributes.txt and fare_rules.txt. <br><br>Conditionally Forbidden:<br>- **Forbidden** if [fare_rules.txt](fare_rulestxt) or [fare_attributes.txt](fare_attributestxt) are defined.<br>- Optional otherwise. |
-|  [fare_transfer_rules.txt](#fare_transfer_rulestxt)  | **Conditionally Forbidden** | Fare rules for transfers between legs of travel.<br><br>Along with fare_leg_rules.txt, file fare_transfer_rules.txt provides a more detailed method for modeling fare structures. As such, the use of fare_transfer_rules.txt is entirely seperate from files fare_attributes.txt and fare_rules.txt. <br><br>Conditionally Forbidden:<br>- **Forbidden** if [fare_rules.txt](fare_rulestxt) or [fare_attributes.txt](fare_attributestxt) are defined.<br>- Optional otherwise. |
+|  [fare_attributes.txt](#fare_attributestxt)  | Optional | Fare information for a transit agency's routes. |
```

Comment:
>  fare_rules.txt doesn't make sense without fare_attributes.txt 

Historically, files `fare_rules.txt` and `fare_attributes.txt` have both been listed as optional. But I agree with your statement. 

Perhaps we could clean this up by keeping `fare_attributes.txt` as optional, and make `fare_rules.txt` conditionally required on `fare_attributes.txt` (since `fare_rules.fare_id` is a required foreign ID referencing `fare_attributes.fare_id`). Thoughts?

Reply:
As for defining relations, perhaps we can add a note in the `fare_attributes.txt` and `fare_leg_rules.txt` file descriptions:

> There are two modelling options for describing fares. GTFS-Fares V1 is the legacy option for describing minimal fare information. GTFS-Fares V2 is an updated method that allows for a more detailed account of an agency's fare structure. Both are allowed to be present in a dataset, but only one method should be used by a data consumer for a given dataset. It is recommended that GTFS-Fares V2 takes precedence over GTFS-Fares V1. <br><br>GTFS-Fares V1: <br>-`fare_attributes.txt`<br>-`fare_rules.txt`<br><br>GTFS-Fares V2: <br>-`fare_leg_rules.txt`<br>-`fare_transfer_rules.txt`

Thoughts?

Reply:
+1 for making fare_rules.txt conditionally required on fare_attributes.txt, though either my English tricks me or probably you meant it to be the other way around. What I think the meaning should be (I leave it to the English speakers to decide how it's clearly written) is that: there can't be fare_rules.txt without fare_attributes.txt (because as you said it has to have foreign keys to fare_attributes.fare_id). There can be fare_attributes.txt without fare_rules.txt (i.e: simple flat fare for the whole agency).

As for the relation between the V1 and V2: all what you wrote is clear and I support the change. But I drop in 1 more thought that we need to address IMHO: what should a producer produce?
- If their real-world fare rules are simple enough to be represented in V1, then should he make V2? Probably not
- If the producer can't represent the reality in V1, then they shouldn't have V1? Or they should possibly have the "closest" "good enough" representation of V1 also added to provide backwards compatibility?

Reply:
I would think this is up to the producer for how exhaustive they want to be. There are a number of agencies now that use V1 to model only what they can of their fare structure, but not exhaustively (as a limit of the model). I would give the liberty to the agency here to decide if they want to be more exhaustive and V2 helps with that.

Reply:
I agree, the liberty should be there, but these things are easy to overlook, Shouldn't we have either a best practice or a note about it?

Regarding the truthiness well, I know too many (of the not too many existing) V1 fares where it's .... lets's call it buggy :) We'll never now if it's lack of the abilities of V1 or the producer :) But you gave a good exampe: if an agency can represent truly the subways in V1 but not the buses or even they can represent the buses but not the complicated transfers between them, then isn't this something we would put in V1 for backwards compatibility?

Reply:
I'm trying to think of a way to recommend V2 while advocating continued use of V1 in the interim and its not coming to me. Recommending a producer to provide both seems demanding, and not a long term solution. IMHO it should be up to producers and consumers to elaborate which version(s) of fares they support.

I'm open to specific suggestions to codify this "note".

Reply:
I don't really understand your last comment. It sounds like we're talking about a B2B contract where the 2 sides agree which format the provider produces. But this is not the case IMHO. Producers automatically generate and publish the GTFS, and consumers consume it. The producer probably doesn't even know most of the consumers. Think about the countless apps. I thought the whole point of GTFS and especially the huge effort mobilitydata invests in these changes is to have more and more publicly available and up-to-date, and precise data.

I agree that if an agency that doesn't have today fares in GTFS starts to produce it after this change passes it makes sense to only provide V2 (aka: it might be not very useful immediately because only a few consumers will be ready to read it, but it doesn't create a regression, because it doesn't exist today, so nobody uses V1). But it's a different situation when an agency has a publicly available GTFS with fares V1 in it. If they would stop producing the V1 a month after the vote passes it would suddenly render this agency's fares non-existant for many consumers. And because there is no B2B relationship we can't even say things like "let's have a 3 month period when we still provide V1 but already provide V2 to give the consumer time to adapt"...

Reply:
I think this is overthinking the transition. The transition will happen naturally and I don't think the spec should dictate the best way to do it. 

Consumers should be aware of the transition. If they are not, I think it's their responsibility to be. Producers will not drop fares v1 data until a good amount of consumers will support it. If they do, it will be a conscious choice in doing so.